paymentsheet/src/main/java/com/stripe/android/paymentsheet/BasePaymentMethodsListFragment.kt

```diff
@@ -75,89 +96,54 @@ internal abstract class BasePaymentMethodsListFragment(
                 color = Color(appearance.getColors(context.isSystemDarkTheme()).appBarIcon),
                 fontFamily = appearance.typography.fontResId
             )
-            isVisible = adapter.hasSavedItems()
-        }
-    }
-
-    @Deprecated("Deprecated in Java")
-    override fun onOptionsItemSelected(item: MenuItem): Boolean {
-        return when (item.itemId) {
-            R.id.edit -> {
-                isEditing = !isEditing
-                true
-            }
-            else -> super.onOptionsItemSelected(item)
         }
     }
 
-    override fun onSaveInstanceState(outState: Bundle) {
-        outState.putBoolean(IS_EDITING, isEditing)
-        super.onSaveInstanceState(outState)
-    }
-
     private fun setupRecyclerView(viewBinding: FragmentPaymentsheetPaymentMethodsListBinding) {
-        val layoutManager = object : LinearLayoutManager(
-            activity,
-            HORIZONTAL,
-            false
-        ) {
-            var canScroll = true
-
-            override fun canScrollHorizontally(): Boolean {
-                return canScroll && super.canScrollHorizontally()
-            }
-        }.also {
+        layoutManager = PaymentMethodsLayoutManager(requireContext()).also {
             viewBinding.recycler.layoutManager = it
         }
 
         adapter = PaymentOptionsAdapter(
-            lpmRepository = sheetViewModel.lpmResourceRepository.getRepository(),
-            canClickSelectedItem = canClickSelectedItem,
-            paymentOptionSelected = ::onPaymentOptionsItemSelected,
-            paymentMethodDeleteListener = ::deletePaymentMethod,
-            addCardClickListener = ::transitionToAddPaymentMethod
+            nameProvider = {
+                val paymentMethod = sheetViewModel.lpmResourceRepository.getRepository().fromCode(it)
+                resources.getString(paymentMethod?.displayNameResource!!)
+            },
```

Comment:
This is a replacement for passing in the entire `LpmRepository`. I’ll look for a way to avoid this entirely and pass in the resolved strings directly.